nilearn/decoding/tests/test_fista.py

```diff
@@ -48,24 +51,35 @@ def test_input_args_and_kwargs():
     y = sig + noise_std * rng.randn(*sig.shape)
     X = np.eye(p)
     mask = np.ones((p,)).astype(bool)
-    alpha = .01
+    alpha = 0.01
     alpha_ = alpha * X.shape[0]
-    l1_ratio = .2
+    l1_ratio = 0.2
     l1_weight = alpha_ * l1_ratio
-    f1 = lambda w: _squared_loss(X, y, w, compute_grad=False)
-    f1_grad = lambda w: _squared_loss(X, y, w, compute_grad=True,
-                                      compute_energy=False)
-    f2_prox = lambda w, l, *args, **kwargs: (_prox_l1(w, l * l1_weight),
-                                             dict(converged=True))
-    total_energy = lambda w: f1(w) + l1_weight * np.sum(np.abs(w))
+    f1 = lambda w: _squared_loss(X, y, w, compute_grad=False)  # noqa: E731
+    f1_grad = lambda w: _squared_loss(  # noqa: E731
+        X, y, w, compute_grad=True, compute_energy=False
+    )
+    f2_prox = lambda w, l, *args, **kwargs: (  # noqa: E731, E741
```

Comment:
probably need a better variable name

Reply:
avoid lambda

Reply:
Sorry, why should we avoid lambda here (and in the next lines) ?

Reply:
PEP8 E731

https://peps.python.org/pep-0008/#programming-recommendations

Always use a def statement instead of an assignment statement that binds a lambda expression directly to an identifier:

# Correct:
```
def f(x): return 2*x
```

# Wrong:
```
f = lambda x: 2*x
```

The first form means that the name of the resulting function object is specifically ‘f’ instead of the generic ‘<lambda>’. This is more useful for tracebacks and string representations in general. The use of the assignment statement eliminates the sole benefit a lambda expression can offer over an explicit def statement (i.e. that it can be embedded inside a larger expression)


see also:
https://stackoverflow.com/questions/25010167/e731-do-not-assign-a-lambda-expression-use-a-def